drivers/sensors/genx320.c

```diff
@@ -581,6 +589,106 @@ static int ioctl(omv_csi_t *csi, int request, va_list ap) {
             }
             break;
         }
+        case OMV_CSI_IOCTL_GENX320_READ_EVENTS: {
+            // Set event mode if not already set.
+            ret = set_event_mode(csi, true);
+            if (ret < 0) {
+                break;
+            }
+
+            if (is_valid_framesize(csi->framesize, true) < 0) {
```

Comment:
If I'm following this correctly, you should probably check here that the frame size is >= the ndarray size.

Reply:
The py_csi_ng layer checks that the array matches the ndarray size. So, you get a nice error message about the array size being wrong.

> Note that the ioctl could just set the frame size internally, based on the ndarray size, which would make it impossible for the user to set one res and pass a smaller ndarray. Though I'm torn between that and staying consistent with all the other scripts.

Yeah, it's possible, but, would need to make sure the ndarray passed a list of qualifications in the py_csi_ng layer. Like, I have to keep the dma sizes, line buffers, and etc. in mind for the frame sizes. Not impossible to handle though, but, it kinda means you aren't passing a frame size and pixformat anymore which makes things inconsistent with other scripts.

Reply:
Ah I see it's validate in csi.

> makes things inconsistent with other scripts as you noticed.
Yeah, let's keep it this way.